frontend/src/components/Card.tsx

```diff
@@ -99,78 +98,76 @@ const Card = ({
-              <div className="flex max-w-4xl items-start justify-start pt-3">
-                {languages && (
-                  <div id="languages" className="flex flex-wrap items-center justify-normal gap-2">
-                    {languages &&
-                      languages
-                        .slice(0, visibleLanguages)
-                        .map((topic, index) => (
-                          <TopicBadge
-                            key={topic || `language-${index}`}
-                            topic={topic}
-                            tooltipLabel={`This repository uses ${topic}`}
-                            type="language"
-                          />
-                        ))}
-                    {languages && languages.length > 18 && (
-                      <button
-                        onClick={loadMoreLanguages}
-                        className="text-gray-600 dark:text-gray-300"
-                      >
-                        {toggleLanguages ? 'Show more' : 'Show less'}
-                      </button>
-                    )}
-                  </div>
-                )}
-                {topics && topics.length > 0 && (
-                  <div id="topics" className="flex flex-wrap items-center justify-normal gap-2">
-                    {topics &&
-                      topics
-                        .slice(0, visibleTopics)
-                        .map((topic, index) => (
-                          <TopicBadge
-                            key={topic || `topic-${index}`}
-                            topic={topic}
-                            tooltipLabel={`This project is labeled as "${topic}"`}
-                            type="topic"
-                          />
-                        ))}
-
-                    {topics && topics.length > 18 && (
-                      <button onClick={loadMoreTopics} className="text-gray-600 dark:text-gray-300">
-                        {toggleTopics ? 'Show more' : 'Show less'}
-                      </button>
-                    )}
-                  </div>
-                )}
-                {social && social.length > 0 && (
-                  <div id="social" className="flex items-center justify-normal gap-2">
-                    {social &&
-                      social.map((item) => (
-                        <a
-                          key={`${item.title}-${item.url}`}
-                          href={item.url}
-                          target="_blank"
-                          rel="noopener noreferrer"
-                          className="flex items-center justify-center gap-1"
-                        >
-                          <FontAwesomeIcon icon={item.icon as FontAwesomeIconProps['icon']} />
-                        </a>
-                      ))}
-                  </div>
-                )}
-              </div>
+          {(languages || topics || social) && (
```

Comment:
Also something I noticed. When there are a lot of labels on mobile view they are sometimes too close to the right border of the card. Could this be updated?
![Screenshot 2024-12-21 at 6 32 56 PM](https://github.com/user-attachments/assets/5edbc91b-4ba3-453f-ae1b-0751d3b0a2a6)

Reply:
Can we sort the labels/topics based on their length and then arrange them to balance the alignment? This can solve the issue of labels being close to the right border of the card.

Before:
![image](https://github.com/user-attachments/assets/70d9a1d0-6226-4ea6-a87c-1ef86d1f6d9d)

After:
![image](https://github.com/user-attachments/assets/d34d98df-b0a3-490a-9ab2-bc3aec22c0f7)